lightning/src/util/persist.rs

```diff
@@ -762,17 +762,18 @@ where
 
 	fn archive_persisted_channel(&self, funding_txo: OutPoint) {
 		let monitor_name = MonitorName::from(funding_txo);
-		let monitor = match self.read_monitor(&monitor_name) {
+		let monitor_key = monitor_name.as_str().to_string();
+		let monitor = match self.read_channel_monitor_with_updates(monitor_key) {
 			Ok((_block_hash, monitor)) => monitor,
```

Comment:
Pre-existing, but I think the `read_monitor` method could use a rename and maybe some docs since it basically reads an old monitor, which seems worth a call-out. No need to do it in this PR though